src/firefly/js/ui/DownloadOptionsDialog.jsx

```diff
@@ -28,152 +28,138 @@ export function getTypeData(key, val='', tip = '', labelV='', labelW) {
-            const wsList = getWorkspaceList();
-            const loc = this.props.fromGroupKey && getFieldVal(this.props.fromGroupKey, 'fileLocation');
-            const wsSelect = this.props.fromGroupKey && getFieldVal(this.props.fromGroupKey, 'wsSelect');
-
-            this.setState((state) => {
-                state = Object.assign({}, state);
-                if (loc !== state.where) {
-                    state.where = loc;
-                    if (loc ===  WORKSPACE) {
-                        state.isUpdating = true;
-                        dispatchWorkspaceUpdate();
-                        return state;
-                    }
-                }
-
-                if (isUpdating !== state.isUpdating) {
-                    state.isUpdating = isUpdating;
-                }
-                if (wsList !== state.wsList) {
-                    state.wsList = wsList;
-                }
-
-                if (wsSelect !== state.wsSelect) {
-                    state.wsSelect = wsSelect;
-                }
-
-                return state;
-            });
+    useEffect(() => {
+        if (wsSelect !== wsSelectTemp) {
+            setWsSelect(wsSelectTemp);
         }
-    }
-
-
-    render() {
-        const {where, wsSelect, wsList, isUpdating} = this.state;
-        const {children, dialogWidth=500, dialogHeight=300} = this.props;
-        const showWorkspace = () => {
-
-            const loading = () => {
-                return (
-                    <div style={{width: '100%', height: '100%', display:'flex', justifyContent: 'center', alignItems: 'center'}}>
-                        <img style={{width:14,height:14}} src={LOADING}/>
-                    </div>
-                );
-            };
-
-            const showSave = () => {
-                return (
-                    <div style={{marginTop: 10,
-                                 boxSizing: 'border-box',
-                                 width: 'calc(100%)', height: 'calc(100% - 10px)',
-                                 overflow: 'auto',
-                                 padding: 5,
-                                 border:'1px solid #a3aeb9'
-                                 }}>
-                        <WorkspaceSave fieldKey={'wsSelect'}
-                                       files={wsList}
-                                       value={wsSelect}
-                        />
-                    </div>
-                );
-            };
-
-            const showNoWSFiles = (message) => {
-                return (
-                    <div style={{marginTop: 10,
-                                 padding: 10,
-                                 boxSizing: 'border-box',
-                                 width: 'calc(100%)',
-                                 textAlign: 'center',
-                                 border:'1px solid #a3aeb9'}}>
-                        {message}
-                    </div>
-                );
-            };
+    }, [wsSelectTemp]);
```

Comment:
Using separate effects are fine. It is you call, but you could also so something like this:

```js
    useEffect(() => {
        if (isUpdating !== isUpdatingTemp) setIsUpdating(isUpdatingTemp);
        if (wsList !== wsListTemp) setWsList(wsListTemp);
        if (wsSelect !== wsSelectTemp) setWsSelect(wsSelectTemp);
    }, [wsSelectTemp,isUpdatingTemp,wsListTemp]);
```
In general we don't want to load too much into an effect because if can become difficult to follow and support. However, this one is so straight forward it would be OK to combine.